### PR TITLE
Update end-to-end with Firefly

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: https://github.com/runtimeverification/firefly/issues/662

### CODE REVIEW

**Comment by ehildenb** [comment]
Let's run the Firefly tool on the example repo here, not on the Firefly repo. We may need to change the `firefly` script a bit to get it to work in other directories.

**Comment by gtrepta** [comment]
How do you mean, do you want me to add an example HelloWorld truffle repo to this one?

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -14,18 +14,26 @@ pipeline {
...
+      environment { GITHUB_TOKEN = credentials('rv-jenkins') }
       when { changeRequest() }
       stages {
+        stage('Checkout Firefly') {
+          steps {
+            dir('firefly') {
+              git url: 'git@github.com:runtimeverification/firefly.git'
+            }
+          }
+        }
```
Comment (ehildenb) [inline jenkins/Jenkinsfile]:
```suggestion
        stage('Checkout Firefly') { steps { dir('firefly') { git url: 'git@github.com:runtimeverification/firefly.git' } } }
```

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -14,18 +14,26 @@ pipeline {
     }
     stage('Build and Test') {
       agent { dockerfile { dir "jenkins" } }
+      environment { GITHUB_TOKEN = credentials('rv-jenkins') }
```
Comment (ehildenb) [inline jenkins/Jenkinsfile]: Do we need this anymore if we're using the `git` Jenkins command to checkout Firefly repo?

Comment (gtrepta) [reply jenkins/Jenkinsfile]: If the git jenkins command works without it then we can get rid of it

File: firefly-run.sh [hunk 3]
```diff
@@ -4,44 +4,7 @@ set -euo pipefail
...
-./kevm web3-send "$PORT" 'firefly_shutdown'
-cd ..
-
-# post the reports
-curl -X POST -F access-token="$FIREFLY_TOKEN"                                        \
-             -F 'status=pass'                                                        \
-             -F 'file=@report.txt'                                                   \
-             -F 'file2=@coverage.json'                                               \
-             -F 'file3=@compiled.zip' 'https://sandbox.fireflyblockchain.com/report'
+cd firefly
```
Comment (ehildenb) [inline firefly-run.sh]:
```suggestion
export PATH=./firefly:$PATH
firefly prep ...
```

**Comment by gtrepta** [comment]
This next build should upload the coverage report with a new token I just created, and I made a github hook for `firefly-demo` on the fireflyblockchain site.

**Review by ehildenb — APPROVED** [review]

**Comment by ehildenb** [comment]
Can we update the settings in Firefly UI to allow lower coverage (just to make sure we can)?

**Pull request merged**